x-pack/plugins/ecs_data_quality_dashboard/server/routes/results/get_results.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter, Logger } from '@kbn/core/server';
+
+import { RESULTS_ROUTE_PATH, INTERNAL_API_VERSION } from '../../../common/constants';
+import { buildResponse } from '../../lib/build_response';
+import { buildRouteValidation } from '../../schemas/common';
+import { GetResultQuery } from '../../schemas/result';
+import type { Result, ResultDocument } from '../../schemas/result';
+import { API_DEFAULT_ERROR_MESSAGE } from '../../translations';
+import type { DataQualityDashboardRequestHandlerContext } from '../../types';
+import { createResultFromDocument } from './parser';
+import { API_RESULTS_INDEX_NOT_AVAILABLE } from './translations';
+
+const getQuery = (patterns: string) => ({
+  size: 0,
+  query: {
+    bool: { filter: [{ terms: { 'rollup.pattern': patterns.split(',') } }] },
+  },
+  aggs: {
+    latest: {
+      terms: { field: 'rollup.pattern', size: 1000000 }, // big enough to get all patterns
+      aggs: { latest_doc: { top_hits: { size: 1, sort: [{ '@timestamp': { order: 'desc' } }] } } },
+    },
+  },
+});
+
+export interface LatestAggResponseBucket {
+  key: string;
+  latest_doc: { hits: { hits: Array<{ _source: ResultDocument }> } };
+}
+
+export const getResultsRoute = (
+  router: IRouter<DataQualityDashboardRequestHandlerContext>,
+  logger: Logger
+) => {
+  router.versioned
+    .get({
+      path: RESULTS_ROUTE_PATH,
+      access: 'internal',
+    })
+    .addVersion(
+      {
+        version: INTERNAL_API_VERSION,
+        validate: { request: { query: buildRouteValidation(GetResultQuery) } },
+      },
+      async (context, request, response) => {
+        const services = await context.resolve(['core', 'dataQualityDashboard']);
+        const resp = buildResponse(response);
+
+        let index: string;
+        try {
+          index = await services.dataQualityDashboard.getResultsIndexName();
+        } catch (err) {
+          logger.error(`[GET results] Error retrieving results index name: ${err.message}`);
+          return resp.error({
+            body: `${API_RESULTS_INDEX_NOT_AVAILABLE}: ${err.message}`,
+            statusCode: 503,
+          });
+        }
+
+        try {
+          const query = { index, ...getQuery(request.query.patterns) };
+          const esClient = services.core.elasticsearch.client.asInternalUser;
```

Comment:
Consider `client.asCurrentUser` such that reads use the permissions of the authenticated user